dbms/src/Storages/Transaction/SSTReader.cpp

```diff
@@ -32,6 +32,10 @@ bool MonoSSTReader::remained() const
         auto key = buffToStrView(proxy_helper->sst_reader_interfaces.fn_key(inner, type));
         if (!end.empty() && key >= end)
         {
+            if (key >= end)
```

Comment:
No, this will only happen in a raftstore v2 cluster. It will happen only once for a SST file.
However, `remained` may be abused since it is not considered and declared to be a cost function which prints. So we'd better have a protection.

Reply:
Yes, I make a mistake here